convert/src/label_studio_to_badgerdoc/label_studio_to_badgerdoc_use_case.py

```diff
@@ -0,0 +1,257 @@
+    AnnotationConverter,
+)
+from src.label_studio_to_badgerdoc.badgerdoc_format.badgerdoc_format import (
+    BadgerdocFormat,
+)
+from src.label_studio_to_badgerdoc.badgerdoc_format.pdf_renderer import (
+    PDFRenderer,
+)
+from src.label_studio_to_badgerdoc.badgerdoc_format.plain_text_converter import (
+    TextToBadgerdocTokensConverter,
+)
+from src.label_studio_to_badgerdoc.models.label_studio_models import (
+    LabelStudioModel,
+    S3Path,
+)
+from src.logger import get_logger
+
+LOGGER = get_logger(__file__)
+
+
+class LabelStudioToBDConvertUseCase:
+    def __init__(
+        self,
+        s3_client,
+        current_tenant: str,
+        token_data: TenantData,
+        s3_input_annotation: S3Path,
+        s3_output_bucket: str,
+        s3_output_file_id: int,
+    ) -> None:
+        page_border_offset = DEFAULT_PAGE_BORDER_OFFSET
+        self.plain_text_converter = TextToBadgerdocTokensConverter(
+            page_border_offset=page_border_offset
+        )
+        self.pdf_renderer = PDFRenderer(page_border_offset=page_border_offset)
+        self.annotation_converter = AnnotationConverter()
+        self.badgerdoc_format = BadgerdocFormat()
+        self.s3_client = s3_client
+
+        self.current_tenant = current_tenant
+        self.token_data = token_data
+        self.request_headers = {
+            "X-Current-Tenant": self.current_tenant,
+            "Authorization": f"Bearer {self.token_data.token}",
+        }
+        self.s3_input_annotation = s3_input_annotation
+        self.s3_output_bucket = s3_output_bucket
+        self.s3_output_file_id = s3_output_file_id
+
+    def execute(self) -> None:
+        label_studio_format = self.download_label_studio_from_s3(
+            self.s3_input_annotation
+        )
+        LOGGER.debug(f"label studio format {label_studio_format=}")
+        self.badgerdoc_format.convert_from_labelstudio(label_studio_format)
+        LOGGER.debug("Tokens and annotations are converted")
+        importjob_id_created = (
+            self.import_annotations_to_annotation_microservice()
+        )
+        self.upload_badgerdoc_format_to_s3(importjob_id_created)
+        LOGGER.debug("Tokens and annotations uploaded")
+
+    def download_label_studio_from_s3(
+        self,
+        s3_input_annotation: S3Path,
+    ) -> LabelStudioModel:
+        with tempfile.TemporaryDirectory() as tmp_dirname:
+            tmp_dirname = Path(tmp_dirname)
+            input_file = tmp_dirname / Path(s3_input_annotation.path).name
+
+            try:
+
+                self.s3_client.download_file(
+                    s3_input_annotation.bucket,
+                    s3_input_annotation.path,
+                    str(input_file),
+                )
+
+            except ClientError as e:
+                raise HTTPException(
+                    status_code=404,
+                    detail=f"Request to download label studio file failed - {e}",
+                )
```

Comment:
Do not expose exception message to outside, it may contain sensitive data. Just log an error
```suggestion
            except ClientError as e:
                LOGGER.exception("Cannot download file from S3: %s", e_g_file_path)
                raise HTTPException(
                    status_code=404,
                    detail="Error during file fetching: file not found",
                ) from err
```

Reply:
Agreed